### ISSUE TITLE
Emacs 30 org-timeblock "compat-macs not found"

### ISSUE THREAD

**Issue by Askath** [ichernyshovvv/org-timeblock#65]
Hello,

Just wanted to mention that I tried this package out with Emacs 30 and noticed that when running org-timeblock I get the error that "compat-macs" is not found, which is required in the org-timeblock.el

I commented the require and the compat-defun out and it seems to run then. (not sure what I broke with that tho)

**Comment by technician77** [comment]
Had same problem using Emacs-NT version 30.0.92.

My solution was to download compat from elpa, extract compat-macs, put it in the software directory where the rest of the built-in compat package is and then delete the elpa version of compat again.

**Comment by alexmurray** [comment]
Since emacs 30 is now in pretest more users will start hitting this issue - any chance it can be worked on?

**Comment by ichernyshovvv** [comment]
@alexmurray 
> Since emacs 30 is now in pretest more users will start hitting this issue - any chance it can be worked on?

Yeah, sometime in the future. Or not (because I might write and continue to develop a general calendar, not specific for org, but for iCalendar and other data also). I don't have neither time or desire to work on it now. PRs are welcome. The issue's not so difficult to solve.

**Comment by ichernyshovvv** [comment]
Or you can live in Emacs 29 for a time

**Comment by Askath** [comment]
I would not mind working on it. What would be a solution to fix this? 
Add compat-emacs to this repo?

**Comment by vermiculus** [comment]
I certainly sympathize with the lack of time/energy/impetus. I had the same limitations with magithub – particularly after the first versions of forge came out. Even so…

> The issue's not so difficult to solve.

Unfortunately it's not clear to me what that solution looks like to you. It might seem obvious, but some brief guidance for those unfamiliar with compat would be helpful.

> Add compat-macs to this repo?

I would assume this is *not* the desirable solution since compat-macs should continue to be maintained in compat.

---

I'm not too familiar with compat (I've never had cause to use it – all I know so far are what's described in the file headers), but I suspect the desired solution here is to upstream the appropriate compat-defun definition below and then to depend on _that_ instead. It looks like `compat-macs` is completely compiled out of emacs30 such that it isn't available to be loaded.

```elisp
(compat-defun org-fold-show-context (&optional key)
  "Make sure point and context are visible."
  (org-show-context key))
```

**Comment by minad** [comment]
I am one of the Compat co-maintainers. compat-macs.el is an internal feature of Compat and must not be used by other packages. See this warning:

https://github.com/emacs-compat/compat/blob/cccd41f549fa88031a32deb26253b462021d7e12/compat-macs.el#L20-L26

Instead of `compat-defun` use conventional `defun` with a `static-if` version check.

**Comment by tsmacdonald** [comment]
If you're on Emacs >= 29.1, are a Philistine, and just want org-timeblock to work right now, one workaround is to open `org-timeblock.el`, remove the `requires` for `compat` and `compat-macs`, remove them from the `Package-Requires` at the top:

```lisp
;; Package-Requires: ((emacs "28.1") (org "9.0") (svg "1.1"))
```

and then modify their usage around L360:

```lisp
; (compat-version "29.1")

; (compat-defun org-fold-show-context (&optional key)
;	      "Make sure point and context are visible."
;	      (org-show-context key))

(defun org-timeblock-show-context ()
  "Make sure point and context are visible."
  (org-fold-show-context 'agenda))
```

(I'm not opening a PR with the `static-if` solution because I don't have a good setup for testing it across versions)

**Comment by Thuruv** [comment]
> If you're on Emacs >= 29.1, are a Philistine, and just want org-timeblock to work right now, one workaround is to open `org-timeblock.el`, remove the `requires` for `compat` and `compat-macs`, remove them from the `Package-Requires` at the top:
> 
> ;; Package-Requires: ((emacs "28.1") (org "9.0") (svg "1.1"))
> and then modify their usage around L360:
> 
> ; (compat-version "29.1")
> 
> ; (compat-defun org-fold-show-context (&optional key)
> ;	      "Make sure point and context are visible."
> ;	      (org-show-context key))
> 
> (defun org-timeblock-show-context ()
>   "Make sure point and context are visible."
>   (org-fold-show-context 'agenda))
> (I'm not opening a PR with the `static-if` solution because I don't have a good setup for testing it across versions)

Got it working., in-addition to the above, comment out the below lines too

``` lisp
;; (compat-version "29.1")

;; (compat-defun org-fold-show-context (&optional key)
;;   "Make sure point and context are visible."
;;   (org-show-context key))

;; (defun org-timeblock-show-context ()
;;   "Make sure point and context are visible."
;;   (compat-call org-fold-show-context 'agenda))
```
